apps/src/studio/Item.js

```diff
@@ -358,3 +444,37 @@ Item.prototype.moveToNextPosition = function () {
   this.x = next.x;
   this.y = next.y;
 };
+
+/**
+ * Mark that we're colliding with object represented by key.
+ * Here, override base implemention to special on-collision logic for certain
+ * item classes.
+ * @param key A unique key representing the object we're colliding with
+ * @returns {boolean} True if collision is started, false if we're already colliding
+ * @override
+ */
+Item.prototype.startCollision = function (key) {
+  var newCollisionStarted = Item.superPrototype.startCollision.call(this, key);
+  if (newCollisionStarted) {
+    if (this.className === 'hazard') {
+      var actor = Studio.sprite[key];
+      if (actor) {
```

Comment:
Low priority: It's a little odd to me that the Item class is aware of the outside world of Studio.sprite and has built-in logic to deal with them. The `autoDisappear` case is handled a little differently (where the controlling logic is aware of sprites, items, and projectiles and understands how they relate). I suppose this trend already started with the roam/flee activity modes and has continued here with watchActor. But it is something to consider for future code organization.